samples/pubsub/workerpool/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"gocloud.dev/internal/workerpool"
+	"gocloud.dev/pubsub"
+	"gocloud.dev/pubsub/mempubsub"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// top is a topic that notifies when a video has been uploaded.
+	// sub is a subscription to this topic.
+	top := mempubsub.NewTopic()
+	sub := mempubsub.NewSubscription(top, time.Hour)
+	defer top.Shutdown(ctx)
+	defer sub.Shutdown(ctx)
+
+	var eg errgroup.Group
+	eg.Go(func() error { return publish(ctx, top) })
+	eg.Go(func() error { return subscribe(ctx, sub) })
+	if err := eg.Wait(); err != nil {
+		log.Printf("workerpool sample: %v", err)
+	}
+}
+
+// publish simulates users uploading videos.
+func publish(ctx context.Context, top *pubsub.Topic) error {
+	for {
+		videoURL := fmt.Sprintf("acmestorage://videos/%d.mpeg", rand.Int())
+		if err := top.Send(ctx, &pubsub.Message{Body: []byte(videoURL)}); err != nil {
+			return fmt.Errorf("sending video upload message: %v", err)
+		}
+		// Simulate some delay between videos being uploaded on an up-and-coming video site.
+		time.Sleep(time.Duration(rand.Intn(10) * 100 * int(time.Millisecond)))
+	}
+	return nil
+}
+
+// subscribe pulls from the subscription and simulates processing the videos.
+func subscribe(ctx context.Context, sub *pubsub.Subscription) error {
+	workerpool.Run(ctx, 10, func(ctx context.Context) workerpool.Task {
+		m, err := sub.Receive(ctx)
+		if err != nil {
+			log.Fatalf("receiving video upload message: %v", err)
+		}
+		videoURL := string(m.Body)
+		return func(ctx context.Context) {
+			log.Printf("Processing %v", videoURL)
+			// Simulate transcoding the video and storing it in various formats.
+			time.Sleep(time.Duration(rand.Intn(10) * int(time.Second)))
+			log.Printf("Done processing %v", videoURL)
+			m.Ack()
+		}
+	})
+	ctx := context.Background()
```

Comment:
I don't understand this line.

Reply:
I don't know how it got there. Removed it.